src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs

```diff
@@ -9,12 +10,680 @@ public class PropertiesShouldNotBeWriteOnlyTests : DiagnosticAnalyzerTestBase
+{
+    public class ClassWithReadableProperty
+    {
+        protected string field;
+        protected string AccessibleProperty2
+        {
+        get { return field; }
+        private set { field = value; }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_internal_Read_private_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        internal string AccessibleProperty3
+        {
+            get { return field; }
+            private set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_protected_internal_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        protected internal string AccessibleProperty4
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_internal_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty5
+        {
+            get { return field; }
+            internal set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_Read_protected_Write()
+        {
+            var code = @"
+using System;
+namespace DesignLibrary
+{
+    public class GoodClassWithReadWriteProperty
+    {
+        protected string field;
+        public string AccessibleProperty6
+        {
+            get { return field; }
+            protected set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+        [Fact]
+        public void CSharp_CA1044Good_public_override_Write()
+        {
+            var code = @"
+using System;
+namespace GoodPropertiesShouldNotBeWriteOnlyTests
+{
+    protected string field;
+    public class DerivedClassWithReadableProperty : ClassWithReadableProperty
+    {
+        public override string ReadableProperty1
+        {
+            set { field = value; }
+        }
+    }
+}";
+            VerifyCSharp(code);
+        }
+
+// Cannot get below test to work, not sure where the issue is.
+//        [Fact]
+//        public void CSharp_CA1044Interface_Write()
+//        {
+//            var code = @"
+//using System;
+//namespace GoodPropertiesShouldNotBeWriteOnlyTests
+//{
+//    public interface IInterface
```

Comment:
Do you have any insight as to what’s wrong with the test?  I don’t know enough of the Interface construct in C# so I commented out the code and was going to go back to it.  The rest of the tests worked as expected.  Like I said can I go ahead and remove the commented tests and submit the code without any comments first?

Thanks, Annie
From: Srivatsn Narayanan [mailto:notifications@github.com]
Sent: Friday, April 1, 2016 3:56 PM
To: dotnet/roslyn-analyzers <roslyn-analyzers@noreply.github.com>
Cc: Annie Poh <anniepoh@microsoft.com>
Subject: Re: [dotnet/roslyn-analyzers] CA1044 Implementation and Unit Test (#882)


In src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/PropertiesShouldNotBeWriteOnlyTests.cs<https://github.com/dotnet/roslyn-analyzers/pull/882#discussion_r58278723>:

> +            set { field = value; }

> +        }

> +    }

> +}";

> +            VerifyCSharp(code);

> +        }

> +

> +// Cannot get below test to work, not sure where the issue is.

> +//        [Fact]

> +//        public void CSharp_CA1044Interface_Write()

> +//        {

> +//            var code = @"

> +//using System;

> +//namespace GoodPropertiesShouldNotBeWriteOnlyTests

> +//{

> +//    public interface IInterface

The test is wrong in this case - this code should produce a warning and the test should be expecting a warning.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn-analyzers/pull/882/files/a54cf1c894cf85f7b0568c553801ac18077024da#r58278723>